quickwit-search/src/collector.rs

```diff
@@ -284,6 +284,12 @@ impl Collector for QuickwitCollector {
         &self,
         segment_fruits: Vec<LeafSearchResponse>,
     ) -> tantivy::Result<Self::Fruit> {
+        if segment_fruits.is_empty() {
```

Comment:
These extra lines that are not helping.

Reply:
ah yeah, you are right. Following up logic are safe even if input is empty